src/main/java/com/minecolonies/coremod/colony/Colony.java

```diff
@@ -441,44 +506,323 @@ public int getID()
+            case TAG_FISHERMAN_FISH:
+                this.triggerAchievement(ModAchievements.achievementCatch100Fish);
+                break;
+            case TAG_FARMER_WHEAT:
+                this.triggerAchievement(ModAchievements.achievementHarvest100Wheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                this.triggerAchievement(ModAchievements.achievementHarvest100Potatoes);
+                break;
+            case TAG_FARMER_CARROTS:
+                this.triggerAchievement(ModAchievements.achievementHarvest100Carrots);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                this.triggerAchievement(ModAchievements.achievementPlant100Saplings);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                this.triggerAchievement(ModAchievements.achievementFell100Trees);
+                break;
+            default:
+                break;
         }
-        if (mobKills >= NUM_MOBS_ACHIEVEMENT_FIFTH)
+    }
+    /**
+     * Trigger fourth achievement.
+     * @param statistic the statistic.
+     */
+    private void triggerFourthAchievement(@NotNull String statistic)
+    {
+        switch (statistic)
         {
-            this.triggerAchievement(ModAchievements.achievementKill1000Mobs);
+            case TAG_GUARD_MOBS:
+                this.triggerAchievement(ModAchievements.achievementKill500Mobs);
+                break;
+            case TAG_MINER_ORES:
+                this.triggerAchievement(ModAchievements.achievementMine500Ores);
+                break;
+            case TAG_MINER_DIAMONDS:
+                this.triggerAchievement(ModAchievements.achievementMine500Diamonds);
+                break;
+            case TAG_BUILDER_HUTS:
+                this.triggerAchievement(ModAchievements.achievementBuild500Huts);
+                break;
+            case TAG_FISHERMAN_FISH:
+                this.triggerAchievement(ModAchievements.achievementCatch500Fish);
+                break;
+            case TAG_FARMER_WHEAT:
+                this.triggerAchievement(ModAchievements.achievementHarvest500Wheat);
+                break;
+            case TAG_FARMER_POTATOES:
+                this.triggerAchievement(ModAchievements.achievementHarvest500Potatoes);
+                break;
+            case TAG_FARMER_CARROTS:
+                this.triggerAchievement(ModAchievements.achievementHarvest500Carrots);
+                break;
+            case TAG_LUMBERJACK_SAPLINGS:
+                this.triggerAchievement(ModAchievements.achievementPlant500Saplings);
+                break;
+            case TAG_LUMBERJACK_TREES:
+                this.triggerAchievement(ModAchievements.achievementFell500Trees);
+                break;
+            default:
+                break;
+        }
+    }
+    /**
+     * Trigger fifth achievement.
+     * @param statistic the statistic.
+     */
+    private void triggerFifthAchievement(@NotNull String statistic)
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') The Cyclomatic Complexity of this method "triggerFifthAchievement" is 12 which is greater than 10 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AMethodCyclomaticComplexity)